schema.go

```diff
@@ -552,11 +557,28 @@ func makeStructField(f reflect.StructField) structField {
 					baseType = Int32Type
 				case reflect.Int64:
 					baseType = Int64Type
+				case reflect.Array:
+					length := int(math.Ceil((math.Log10(2) + float64(precision)) / math.Log10(256)))
```

Comment:
We're performing this same computation in two different places, what do you think about moving it to an internal function instead?

Reply:
I went ahead and added a function for this but due to your other suggestion that simplifies the kind checking it is only being used once now. I still like the function suggestion since it allows adding some explanation. Let me know if any issues with the function naming or location.